packages/opentelemetry-node/src/instrumentation/PluginLoader.ts

```diff
@@ -78,6 +78,16 @@ export class PluginLoader {
         return this;
       }
 
+      const alreadyRequiredModules = Object.keys(require.cache);
+      const requiredModulesToHook = modulesToHook.filter(name =>
+        alreadyRequiredModules.some(cached => cached.includes(name))
+      );
+      if (requiredModulesToHook.length > 0) {
+        this.logger.warn(
+          `Some modules (${requiredModulesToHook}) were already required when their respective plugin were loaded. They might not work.`
```

Comment:
In addition to the previous comment, I think some troubleshooting tips could be included so users can proceed with some action item. E.g. "make sure opentelemetry SDK is setup before you require in other modules."